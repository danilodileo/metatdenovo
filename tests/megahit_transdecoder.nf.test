nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_megahit_transdecoder"

    test("-profile test_megahit_transdecoder") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_metatdenovo_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // Important files with unstable content
                    path("${params.outdir}/samtools/megahit.SAMPLE1_PE.flagstat").readLines().size(),
                    path("${params.outdir}/summary_tables/megahit.transdecoder.counts.tsv.gz").linesGzip.findAll { l -> l.contains('orf') },
                    path("${params.outdir}/summary_tables/megahit.transdecoder.overall_stats.tsv.gz").linesGzip.size(),
                    path("${params.outdir}/summary_tables/megahit.transdecoder.overall_stats.tsv.gz").linesGzip.findAll { l -> l.contains('n_feature_count') },
                ).match() },
                { assert path("${params.outdir}/megahit/megahit_out/megahit_assembly.log").readLines().any { l -> l.contains('ALL DONE') } },
                { assert path("${params.outdir}/megahit/megahit_out/megahit_assembly.contigs.fa.gz").linesGzip.size() > 3000 },
                { assert path("${params.outdir}/samtools/megahit.SAMPLE1_PE.idxstats").readLines().size() > 1600 },
                { assert path("${params.outdir}/transdecoder/megahit_assembly.contigs.fa.transdecoder.gff3.gz").linesGzip.findAll { l -> l.contains('CDS') }.size() > 2200 },
                { assert path("${params.outdir}/summary_tables/megahit.transdecoder.counts.tsv.gz").linesGzip.size() > 2000 },
            )
        }
    }
}
