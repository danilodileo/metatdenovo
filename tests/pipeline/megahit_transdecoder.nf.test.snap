{
    "SAMPLE1_PE_1_val_1_fastqc.html": {
        "content": [
            "SAMPLE1_PE_1_val_1_fastqc.html:md5,89c71d5bf2499665b8c036a2ffae3bd7"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:03:34.857244"
    },
    "SAMPLE1_PE.bbmap.log": {
        "content": [
            "SAMPLE1_PE.bbmap.log:md5,13ff97be43d6378ab17a2f5c4d024cf5"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:03:34.879899"
    },
    "megahit_assembly.log": {
        "content": [
            "megahit_assembly.log:md5,b0857ecf10370da5a7eb4e166700b7d9"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:03:34.866892"
    },
    "multiqc_report.html": {
        "content": [
            "multiqc_report.html:md5,9529c3d04e91f3fd009332190c7bb8cb"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:03:34.929466"
    },
    "SAMPLE1_PE.featureCounts.tsv.summary": {
        "content": [
            "SAMPLE1_PE.featureCounts.tsv.summary:md5,5dc1f3dd632dc9cf90aba560c4fa102e"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:03:34.890888"
    }
}